server/services/v1/query_runner.go

```diff
@@ -675,6 +680,46 @@ func (runner *SearchQueryRunner) getFacetFields(collFields []*schema.Field) (qse
 	return facets, nil
 }
 
+func (runner *SearchQueryRunner) getFieldSelection(collFields []*schema.Field) (*read.FieldFactory, error) {
+	var selectionFields []string
+	var include bool
+
+	// Only one of include/exclude. Honor inclusion over exclusion
+	if len(runner.req.IncludeFields) > 0 {
+		selectionFields = runner.req.IncludeFields
+		include = true
+	} else if len(runner.req.ExcludeFields) > 0 {
+		selectionFields = runner.req.ExcludeFields
+		include = false
+	} else {
+		return nil, nil
+	}
+
+	factory := &read.FieldFactory{
+		Include: map[string]read.Field{},
+		Exclude: map[string]read.Field{},
+	}
+
+	for _, sf := range selectionFields {
+		found := false
+		for _, cf := range collFields {
+			if sf == cf.FieldName {
+				found = true
+			}
+		}
+		if !found {
+			return nil, api.Errorf(api.Code_INVALID_ARGUMENT, "`%s` is not a schema field", sf)
+		}
+
+		factory.AddField(&read.SimpleField{
+			Name: sf,
+			Incl: include,
```

Comment:
We can just set `Incl: len(runner.req.IncludeFields) > 0` and remove `include` variable.

Reply:
done